core/src/main/java/org/apache/calcite/rel/core/Collect.java

```diff
@@ -132,6 +133,47 @@ public static Collect create(RelNode input,
             fieldName, input.getRowType()));
   }
 
+  /**
+   * Creates a Collect.
+   *
+   * @param input          Input relational expression
+   * @param sqlKind        SqlKind
+   * @param fieldName      Name of the sole output field
+   */
+  public static Collect create(RelNode input,
```

Comment:
Should we deprecate any of the previous `create` methods to encourage people to use this new one? Or are they all still valid?

Reply:
I think yes, `create(RelNode input,SqlTypeName collectionType, String fieldName)` should be deprecated.

Reply:
Ok, in that case we should not be calling it anymore.
If I am not mistaken, the PR has removed all the previously existing calls (in favor of the new `create` method), but said new method calls the deprecated one (Collect.java:174): it should call the constructor directly in that line.

Reply:
Fixed